src/coreclr/vm/interpexec.cpp

```diff
@@ -83,6 +83,21 @@ static void InterpBreakpoint()
 #define LOCAL_VAR(offset,type) (*LOCAL_VAR_ADDR(offset, type))
 #define NULL_CHECK(o) do { if ((o) == NULL) { COMPlusThrow(kNullReferenceException); } } while (0)
 
+template <typename TResult, typename TSource> void ConvOvfHelper(int8_t *stack, const int32_t *ip)
+{
+    TSource src = LOCAL_VAR(ip[2], TSource);
+    // We have to use `>= and <=`, not `< or >`, in order to behave correctly for NaN and infinities
+    if ((src >= std::numeric_limits<TResult>::min()) && (src <= std::numeric_limits<TResult>::max()))
```

Comment:
I do not think this condition is 100% correct for the boundary cases. It will incorrectly throw exceptions for values that are tiny bit higher/lower than the boundary condition, but that won't overflow once rounded towards zero.

The easiest way to get this right is by duplicating the exact boundaries from https://github.com/dotnet/runtime/blob/1cda0cd64ef617956a7a6276099184a55f61d2da/src/libraries/System.Private.CoreLib/src/System/Math.cs#L1668-L1712